### ISSUE TITLE
Globally registered component naming converts to kebap-case when PascalCase is used

### ISSUE THREAD

**Issue by steffans** [vuejs/vue#7825]
### Version
2.5.15

### Steps to reproduce
Currently all globally registered components automatically convert their name to  kebap-case. 

```js
// register component
Vue.component('RouterLink', {...})

// get component fails, returns undefined
Vue.component('RouterLink')

// get component works, returns the component
Vue.component('router-link')
```

It would be nice to have the registered components also available as with their PascalCase naming. This would also mean that you could use the PascalCase naming in `.vue` templates, which is a nice way to quickly distinguish Vue components from HTML elements. Right now its is not possible, and to make it work this code has to be used:

```html
<template>

  <RouterLink :to="link">Link</RouterLink>

</template>
<script>
{
    components: {
       RouterLink: Vue.component('router-link')
   }
}
</script>
```

### What is expected?
Access globally registered components with their PascalCase naming if it was provided on register function call.

### What is actually happening?
For now the component naming always converts to kebap-case.

**Comment by posva** [comment]
You probably mean the other way around:

```js
Vue.component('FooLink', { template: `<p>Foo</p>` })

!!Vue.component('FooLink') // -> true
!!Vue.component('foo-link') // -> false
```

**Comment by steffans** [comment]
@posva You are right, sorry my bad. I recently tried to use the VueRouter with `RouterLink` which didn't work, looked up the repo a saw `Vue.component('RouterLink', Link)` which got me confused. However this was not the actual released version 3.0.1 i was using, which is still doing `Vue.component('router-link', Link)`.
